### ISSUE TITLE
[Noob Q] How do I use FILTER_REGEX_EXCLUDE?

### ISSUE THREAD

**Issue by KineticTheory** [github/super-linter#847]
Can someone provide an example for me wrt `FILTER_REGEX_EXCLUDE?.  I would like to eliminate this warning:
```console
2020-10-13 15:58:15 [WARN  ]   Failed to get filetype for:[src/linear/qrdcmp.hh]!
```

I currently have this in `linter.yml`, but it doesn't eliminate the warning:
```yaml
jobs:
  build:
    name: Lint Code Base
    runs-on: ubuntu-latest

    steps:
      - name: Checkout Code
        uses: actions/checkout@v2

      - name: Lint Code Base
        uses: github/super-linter@v3
        env:
          DEFAULT_BRANCH: develop
          FILTER_REGEX_EXCLUDE: .*/.*.hh
```

Do I need to `.*/.*/.*.hh`?  The regex format isn't exactly what I am used to using.

Thank you!

**Comment by admiralAwkbar** [comment]
@KineticTheory i think the line should simply be:
```yml
FILTER_REGEX_EXCLUDE: *.hh
```
That would exclude all files that end in `hh`

Now the issue is that, we build the list of files at the beginning and try to find their arrays, and any that don't have a defined home `somefile.hh` will likely still show a warning as it doesn't have a linter or know what it is...

**Comment by KineticTheory** [comment]
> @KineticTheory i think the line should simply be:
> 
> ```yaml
> FILTER_REGEX_EXCLUDE: *.hh
> ```
> 
> That would exclude all files that end in `hh`
> 
> Now the issue is that, we build the list of files at the beginning and try to find their arrays, and any that don't have a defined home `somefile.hh` will likely still show a warning as it doesn't have a linter or know what it is...

Thank you for responding.  If I need multiple filters, should I use a comma delimiter like this `*.hh,*.cpp`?  Does the regex format allow the use of `$` for end-of-line like this `*.hh$,*.cc$`?

**Comment by admiralAwkbar** [comment]
@KineticTheory 
The line of code thats called is:
```bash
[[ ! (${LIST_FILES[$index]} =~ $FILTER_REGEX_EXCLUDE) ]] && unset -v 'LIST_FILES[$index]'
```

So i would assume if you used properly `or` technique, you could get multiple excludes...
@ferrarimarco  would you believe that?

**Comment by KineticTheory** [comment]
> @KineticTheory
> The line of code thats called is:
> 
> ```shell
> [[ ! (${LIST_FILES[$index]} =~ $FILTER_REGEX_EXCLUDE) ]] && unset -v 'LIST_FILES[$index]'
> ```
> 
> So i would assume if you used properly `or` technique, you could get multiple excludes...
> @ferrarimarco would you believe that?

Hmmm.  That line of code would indicate that the value `*.hh` wouldn't work as expected.  I think I need `.*[.]hh`.  Do you concur?

In the end, I think I might need something like `.*[.](cc|hh)$`.  Do I need to escape the parenthesis?  Something like `.*[.]\(cc\|hh\)$`?  I'll give this a try with the understanding that I'll still get the warnings about filetype.

**Comment by ferrarimarco** [comment]
@admiralAwkbar this looks interesting. I'd check #786 as well, which proposes an interesting approach to this.

**Comment by catborise** [comment]
beside that there are also some problems with filter regex
i am using that filter:
FILTER_REGEX_EXCLUDE: .*[static|scss]/.* 
it works as expected but when i add another folder name. then it is not working, not linting and not filtering.

FILTER_REGEX_EXCLUDE: .*[static|scss|venv]/.*

**Comment by jamiekt** [comment]
@KineticTheory Hi, did you ever figure out how to specify multiple excludes?

**Comment by nvuillam** [comment]
Did you try (*.hh|*.cpp) ? :)

**Comment by jamiekt** [comment]
@nvuillam Hi, I did, and it worked great. Should have come back here and updated, apologies.

Specifically, this is what worked for me `(docs|.devcontainer)` https://regex101.com/r/EyHpXc/1

**Comment by KineticTheory** [comment]
Similarly, I am using the following expression to exclude all c++ files and any path that has 'test' in the name.
```yaml
  FILTER_REGEX_EXCLUDE: (.*[.]cc$|/test/)
```
